### PR TITLE
Add v5 upgrade guide

### PR DESCRIPTION
**Pull request by rmccue**
Still in progress, but pushing early.

### CODE REVIEW

File: other-docs/guides/upgrading/v5.md [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+
+In order to set the groundwork for future import/export and migration tooling, GUIDs for post objects in Altis now use UUID URNs (e.g. `urn:uuid:d72fdc38-1305-4568-9821-463120b250f3`). This replaces the former URL-style GUIDs used for posts (e.g. `http://altis.local/?p=42`).
+
+Due to legacy use of the GUID field, some older code may attempt to use the GUID for image media source URLs. This code must be updated to use `wp_get_attachment_url()` instead, which will also ensure that Tachyon and the Altis CDN are correctly used.
+
+New code should only use the GUID field to uniquely identify posts, and never as a URL.
+
+### Minimum version of the AWS SDK
+
+The minimum version of the AWS SDK included with Altis has been bumped to 3.150.0. We recommend keeping the SDK up-to-date via Dependabot.
```
Comment (joehoyle) [inline other-docs/guides/upgrading/v5.md:131]: So we recommend this? Seems like a pain to call out specifically.

**Review by joehoyle — APPROVED** [review]

File: other-docs/guides/upgrading/v5.md (hunk 1 above)
Comment (rmccue) [reply other-docs/guides/upgrading/v5.md:131]: Might drop this section entirely as it seems to be backwards compatible, but had included as an "FYI our minimum of this dep you might also be using is being bumped"

File: other-docs/guides/upgrading/v5.md [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+```
+
+Note that while Altis v4 bundles version 1 of the AMP plugin, version 2 is now available with [many major changes](https://github.com/ampproject/amp-wp/releases/tag/2.0.0), so consider updating at the same time.
+
+If you're using Facebook Instant Articles, you will need to require the FBIA plugin. This requires first [adding WordPress Packagist as a repository](docs://getting-started/third-party-plugins#managing-plugins-via-composer), then requiring the plugin from wpackagist:
+
+```sh
+$ composer require wpackagist-plugin/fb-instant-articles
+```
+
```
Comment (roborourke) [inline other-docs/guides/upgrading/v5.md:55]: If we add core sitemaps in that can go here too

Comment (rmccue) [reply other-docs/guides/upgrading/v5.md:55]: Can handle in follow-up once merged?

File: other-docs/guides/upgrading/v5.md [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+### Suggestions when searching
+
+Search suggestions can now be retrieved from Elasticsearch using autosuggest mode. Autosuggest mode matches partial words rather than whole words, which improves relevance while typing.
+
+Autosuggest mode also matches against a subset of post fields rather than all fields, and defaults to true when the current request is an `admin-ajax.php` request (i.e. `DOING_AJAX === true`).
+
+Consult [the query integration documentation](docs://search/cms-query-integration.md) for further details on how to use and control this functionality.
+
+This feature does not require reindexing, however due to changes in the stemmer, relevance of autosuggest results can be improved by reindexing. This can be [performed at your discretion](docs://search/reindexing-content.md).
```
[the remark this answers is not in the document]
Comment (rmccue) [reply other-docs/guides/upgrading/v5.md]: Did the default config for that change?

[the remark this answers is not in the document]
Comment (roborourke) [reply other-docs/guides/upgrading/v5.md]: No sorry I'm getting my wires crossed - they're actually separate. The `autosuggest` in config is really just for adding the client side search form dropdown now.

This does require reindexing though as it needs an updated mapping.

File: other-docs/guides/upgrading/v5.md [hunk 4]
```diff
@@ -0,0 +1,143 @@
...
+
+The sites list in the Network Admin now displays a more user-friendly view of sites, using site names rather than domains. Additionally, some irrelevant options (such as the mark-as-spam ability) are now de-emphasised in the UI.
+
+### Development Environments
+
+Compatibility issues with using XDebug in Local Server on a Linux host have been fixed. The remote host configuration option is now set automatically for you internally in order to correctly route connections. Consult the [XDebug documentation](docs://local-server/using-xdebug.md) for more information on how to use it in your editor.
+
+### Other Changes
+
+Altis v5 includes many other small changes, consult the changelog for full details of all the changes shipped in this release.
```
Comment (roborourke) [inline other-docs/guides/upgrading/v5.md:143]: GUessing we link those changelogs up once the stable is released?

File: other-docs/guides/upgrading/v5.md [hunk 5]
```diff
@@ -0,0 +1,143 @@
...
+
+### Suggestions when searching
+
+Search suggestions can now be retrieved from Elasticsearch using autosuggest mode. Autosuggest mode matches partial words rather than whole words, which improves relevance while typing.
+
+Autosuggest mode also matches against a subset of post fields rather than all fields, and defaults to true when the current request is an `admin-ajax.php` request (i.e. `DOING_AJAX === true`).
+
+Consult [the search query integration documentation](docs://search/cms-query-integration.md) for further details on how to use and control this functionality.
+
+This feature does not require reindexing, however due to changes in the stemmer, relevance of autosuggest results can be improved by reindexing. [Reindexing can be performed at your discretion](docs://search/reindexing-content.md).
```
Comment (roborourke) [inline other-docs/guides/upgrading/v5.md:91]:
```suggestion
This feature does require reindexing as it makes changes to the field mapping. [Reindexing can be performed at your discretion](docs://search/reindexing-content.md).
```

**Review by roborourke — APPROVED** [review]
Good to go unless you want to remove the bit about the AWS SDK

**Pull request merged**